blockmanager_test.go

```diff
@@ -624,3 +626,46 @@ func TestBlockManagerInvalidInterval(t *testing.T) {
 		)
 	}
 }
+
+// buildOldFilter creates a CFilter with all output scripts except all
+// OP_RETURNS with push-only scripts.
+//
+// NOTE: this is not a valid filter, only for tests.
+func buildOldFilter(block *wire.MsgBlock) (*gcs.Filter, error) {
+	blockHash := block.BlockHash()
+	b := builder.WithKeyHash(&blockHash)
+
+	for _, tx := range block.Transactions {
+		for _, txOut := range tx.TxOut {
+			// The old version of BIP-158 skipped OP_RETURNs that
+			// had a push-only script.
+			if txOut.PkScript[0] == txscript.OP_RETURN &&
+				txscript.IsPushOnlyScript(txOut.PkScript[1:]) {
+				continue
+			}
+
+			b.AddEntry(txOut.PkScript)
+		}
+	}
+
+	return b.Build()
+}
+
+// buildOldOldFilter creates a CFilter with all output scripts, including
+// OP_RETURNS.
+//
+// NOTE: this is not a valid filter, only for tests.
+func buildOldOldFilter(block *wire.MsgBlock) (*gcs.Filter, error) {
```

Comment:
Lol could rename these to what the filter includes. So this'd be something like `buildAllPkScriptsFilter`.

Reply:
lol fixed